sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/SchemaIOTableProviderWrapper.java

```diff
@@ -118,6 +123,34 @@ public Schema getSchema() {
       return begin.apply(readerTransform);
     }
 
+    @Override
+    public PCollection<Row> buildIOReader(
+        PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
+      PTransform<PBegin, PCollection<Row>> readerTransform = schemaIO.buildReader();
+      if (readerTransform instanceof PushdownProjector) {
+        PushdownProjector pushdownProjector = (PushdownProjector) readerTransform;
+        FieldAccessDescriptor fieldAccessDescriptor =
+            FieldAccessDescriptor.withFieldNames(fieldNames);
+        // The pushdown must return a PTransform that can be applied to a PBegin, or this cast will
+        // fail.
+        readerTransform =
+            (PTransform<PBegin, PCollection<Row>>)
+                pushdownProjector.withProjectionPushdown(fieldAccessDescriptor);
+      }
```

Comment:
Shouldn't this have an `else` that looks something like `BaseBeamTable` and reject attempts to push down if it isn't supported? You should probably also reject filter pushdown (the `if (!(filters instanceof DefaultTableFilter)) {` block).

https://github.com/apache/beam/blob/243128a8fc52798e1b58b0cf1a271d95ee7aa241/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/BaseBeamTable.java#L30

Reply:
Done.